lib/ansible/modules/windows/win_shortcut.ps1

```diff
@@ -5,33 +5,49 @@
 
 # Based on: http://powershellblogger.com/2016/01/create-shortcuts-lnk-or-url-files-with-powershell/
 
-#Requires -Module Ansible.ModuleUtils.Legacy
-
-$ErrorActionPreference = "Stop"
+#AnsibleRequires -CSharpUtil Ansible.Basic
+
+$spec = @{
+    options = @{
+        src = @{ type='str' }
+        dest = @{ type='path'; required=$true }
+        state = @{ type='str'; default='present'; choices=@( 'absent', 'present' ) }
+        args = @{ type='str' }
+        directory = @{ type='path' }
+        hotkey = @{ type='str' }
+        icon = @{ type='path' }
+        description = @{ type='str' }
+        windowstyle = @{ type='str'; choices=@( 'maximized', 'minimized', 'normal' ) }
+    }
+    supports_check_mode = $true
+}
 
-$params = Parse-Args -arguments $args -supports_check_mode $true
-$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
 
-$orig_src = Get-AnsibleParam -obj $params -name "src"
-$dest = Get-AnsibleParam -obj $params -name "dest" -type "path" -failifempty $true
-$state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "present" -validateset "absent","present"
-$orig_args = Get-AnsibleParam -obj $params -name "args" -type "str"
-$directory = Get-AnsibleParam -obj $params -name "directory" -type "path"
-$hotkey = Get-AnsibleParam -obj $params -name "hotkey" -type "str"
-$icon = Get-AnsibleParam -obj $params -name "icon" -type "path"
-$orig_description = Get-AnsibleParam -obj $params -name "description" -type "str"
-$windowstyle = Get-AnsibleParam -obj $params -name "windowstyle" -type "str" -validateset "maximized","minimized","normal"
+$src = $module.Params.src
+$dest = $module.Params.dest
+$state = $module.Params.state
+$arguments = $module.Params.args  # Variable $args is a special variable
+$directory = $module.Params.directory
+$hotkey = $module.Params.hotkey
+$icon = $module.Params.icon
+$description = $module.Params.description
+$windowstyle = $module.Params.windowstyle
 
 # Expand environment variables on non-path types
-$args = Expand-Environment($orig_args)
-$description = Expand-Environment($orig_description)
-$src = Expand-Environment($orig_src)
-
-$result = @{
-    changed = $false
-    dest = $dest
-    state = $state
+if ($null -ne $src) {
+    $src = [System.Environment]::ExpandEnvironmentVariables($src)
 }
+if ($null -ne $arguments) {
+    $arguments = [System.Environment]::ExpandEnvironmentVariables($arguments)
+}
+if ($null -ne $description) {
+    $description = [System.Environment]::ExpandEnvironmentVariables($description)
+}
+
+$module.Result.changed = $false
```

Comment:
Just an FYI, changed is automatically set on the return result to `$false`.

Reply:
Makes sense. Fixed.